src/util/safe_print.cpp

```diff
@@ -0,0 +1,200 @@
+/*********************                                                        */
+/*! \file safe_print.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andres Noetzli
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2016 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief [[ Add one-line brief description here ]]
+ **
+ ** [[ Add lengthier description here ]]
+ ** \todo document this file
+ **/
+
+#include "safe_print.h"
+
+#include <unistd.h>
+
+/* Size of buffers used */
+#define BUFFER_SIZE 20
+
+namespace CVC4 {
+
+template <>
+void safe_print(int fd, const char* msg) {
+  ssize_t ret = write(fd, msg, safe_strlen(msg));
```

Comment:
Turns out that there is quite an elegant solution for string literals. With the new function type `template<typename T> void CVC4_PUBLIC safe_print(int fd, const T& msg)`, doing calling `safe_print()` with a string literal calls `safe_print(int fd, const char (&msg)[N])`, so I just defined an additional function of that type (I committed a version of this s.t. you can take a look).
It now looks like `progName` is not used in a statistic, so that is fine. However, `filename` is used in a statistic and it is the only instance of a `const char*` statistic that I could find. The best way to solve this seems to be to turn `filename` into an `std::string` (actually it is currently a `c_str()` of a string when not using stdin). Alternatively, I could wrap it in a string before creating the statistic but that seems kind of pointless. Which way do you prefer?